### PR TITLE
Feature 7 publish date

### LINKED ISSUE #7: Published date
As a user I would like to have published date property in my index for news

### PR DESCRIPTION
**Pull request by anttileppa**
Resolves #7

### CODE REVIEW

File: src/functions/add-category-to-document/handler.ts [hunk 1]
```diff
@@ -27,45 +18,51 @@ enum ContentCategory {
...
+      if ($("time[itemprop='datePublished']").length) {
+        return ContentCategory.NEWS;
+      }
+    }
 
-  if (!element.length) {
-    console.warn(`Could not find drupal-settings-json from ${url} `);
-    return ContentCategory.UNCATEGORIZED;
-  }
+    const element = $("script[data-drupal-selector=drupal-settings-json]");
```
Comment (Daniilmass) [inline src/functions/add-category-to-document/handler.ts:35]:
```suggestion
```
No need for empty line

File: src/functions/add-category-to-document/handler.ts [hunk 2]
```diff
@@ -27,45 +18,51 @@ enum ContentCategory {
...
+    }
 
-  const config: DrupalSettingsJson = JSON.parse(jsonString);
-  if (!config) {
-    console.warn(`Could parse drupal-settings-json from ${url} `);
-    return ContentCategory.UNCATEGORIZED;
-  }
+    const jsonString = element.html();
+    if (!jsonString.length) {
+      console.warn(`Could find drupal-settings-json string from ${url} `);
```
Comment (Daniilmass) [inline src/functions/add-category-to-document/handler.ts:43]:
```suggestion
      console.warn(`Could not find drupal-settings-json string from ${url} `);
```
Should it be "not"?

File: src/functions/add-category-to-document/handler.ts [hunk 3]
```diff
@@ -27,45 +18,51 @@ enum ContentCategory {
...
+    }
 
-  const currentPath = config.path?.currentPath;
-  if (!currentPath) {
-    console.warn(`Could find drupal-settings-json currentPath from ${url} `);
-    return ContentCategory.UNCATEGORIZED;
-  }
+    const config: DrupalSettingsJson = JSON.parse(jsonString);
+    if (!config) {
+      console.warn(`Could parse drupal-settings-json from ${url} `);
```
Comment (Daniilmass) [inline src/functions/add-category-to-document/handler.ts:49]:
```suggestion
      console.warn(`Could not parse drupal-settings-json from ${url} `);
```
Should it be not?

File: src/functions/add-category-to-document/handler.ts [hunk 4]
```diff
@@ -27,45 +18,51 @@ enum ContentCategory {
...
+      return ContentCategory.UNCATEGORIZED;
+    }
 
-  if (currentPath.includes("tpr-unit")) {
-    return ContentCategory.UNIT;
-  }
+    const currentPath = config.path?.currentPath;
+    if (!currentPath) {
+      console.warn(`Could find drupal-settings-json currentPath from ${url} `);
```
Comment (Daniilmass) [inline src/functions/add-category-to-document/handler.ts:55]:
```suggestion
      console.warn(`Could not find drupal-settings-json currentPath from ${url} `);
```
Should it be "not"?

**Review by lauriLukkarinen — APPROVED** [review]

File: src/functions/detect-news-published/handler.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  );
+
+  const updateDocuments = [];
+
+  for (const document of flattenedDocuments) {
+    const publishDate = await detectNewsPublishedDateFromDocument(document);
+    if (publishDate && publishDate.isValid) {
+      updateDocuments.push({ ...document, publish_date: publishDate.toISO() });
+    } else {
+      console.warn("Failed for to detect publish date for document", document);
```
Comment (Daniilmass) [inline src/functions/detect-news-published/handler.ts:109]:
```suggestion
      console.warn("Failed to detect publish date for document", document);
```

**Review by Daniilmass — CHANGES_REQUESTED** [review]

**Review by Daniilmass — APPROVED** [review]

**Pull request merged**